test/suite/test_timestamp08.py

```diff
@@ -0,0 +1,261 @@
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# test_timestamp08.py
+#   Timestamps: multistep transactions
+#
+
+from helper import copy_wiredtiger_home
+import random
+from suite_subprocess import suite_subprocess
+import wiredtiger, wttest
+from wtscenario import make_scenarios
+
+def timestamp_str(t):
+    return '%x' % t
+
+def timestamp_ret_str(t):
+    s = timestamp_str(t)
+    if len(s) % 2 == 1:
+        s = '0' + s
+    return s
+
+class test_timestamp08(wttest.WiredTigerTestCase, suite_subprocess):
+    table_ts_log     = 'table:ts08_ts_logged'
+    table_ts_nolog   = 'table:ts08_ts_nologged'
+
+    types = [
+        ('col_fix', dict(empty=1, extra_config=',key_format=r,value_format=8t')),
+        ('col_var', dict(empty=0, extra_config=',key_format=r')),
+        ('lsm', dict(empty=0, extra_config=',type=lsm')),
+        ('row', dict(empty=0, extra_config='',)),
+    ]
+    ckpt = [
+        ('ckpt_ts_def', dict(ckptcfg='', ckpt_ts=True)),
+        ('ckpt_ts_false', dict(ckptcfg='use_timestamp=false', ckpt_ts=False)),
+        ('ckpt_ts_true', dict(ckptcfg='use_timestamp=true', ckpt_ts=True)),
+    ]
+
+    conncfg = [
+        ('nolog', dict(conn_config='create', using_log=False)),
+        ('V1', dict(conn_config='create,log=(enabled),compatibility=(release="2.9")', using_log=True)),
+        ('V2', dict(conn_config='create,log=(enabled)', using_log=True)),
+    ]
+
+    scenarios = make_scenarios(conncfg, types, ckpt)
+
+    # Check that a cursor (optionally started in a new transaction), sees the
+    # expected values.
+    def check(self, session, txn_config, tablename, expected, prn=False):
+
+        if txn_config:
+            session.begin_transaction(txn_config)
+
+        cur = session.open_cursor(tablename, None)
+        actual = dict((k, v) for k, v in cur if v != 0)
+        if prn == True:
+            print "CHECK : Expected"
+            print expected
+            print "CHECK : Actual"
+            print actual
+        self.assertTrue(actual == expected)
+        # Search for the expected items as well as iterating
+        for k, v in expected.iteritems():
+            self.assertEqual(cur[k], v, "for key " + str(k))
+        cur.close()
+        if txn_config:
+            session.commit_transaction()
+    #
+    # Take a backup of the database and verify that the value we want to
+    # check exists in the tables the expected number of times.
+    #
+    def backup_check(self, check_value, expected_ts_log, expected_ts_nolog):
+
+        newdir = "BACKUP"
+        copy_wiredtiger_home('.', newdir, True)
+
+        conn = self.setUpConnectionOpen(newdir)
+        session = self.setUpSessionOpen(conn)
+        cur_ts_log      = session.open_cursor(self.table_ts_log)
+        cur_ts_nolog    = session.open_cursor(self.table_ts_nolog)
+        # Count how many times the check_value is present in the
+        # logged timestamp table.
+        actual_ts_log = 0
+        for k, v in cur_ts_log:
```

Comment:
Can be done in pythonic way like this sum( x == check_value for x in cur_ts_log.values() )

Reply:
no, as cur_ts_log is not a dict, but a cursor